config-cli/src/main/java/de/adorsys/keycloak/config/service/AuthenticationFlowsImportService.java

```diff
@@ -154,14 +159,149 @@ private void recreateTopLevelFlow(
             AuthenticationFlowRepresentation topLevelFlowToImport,
             AuthenticationFlowRepresentation existingAuthenticationFlow
     ) {
-        AuthenticationManagementResource flowsResource = authenticationFlowRepository.getFlows(realm.getRealm());
-
         AuthenticationFlowRepresentation patchedAuthenticationFlow = CloneUtils.deepPatch(existingAuthenticationFlow, topLevelFlowToImport);
 
-        flowsResource.deleteFlow(patchedAuthenticationFlow.getId());
+        UsedRegistrationFlowWorkaround workaround = new UsedRegistrationFlowWorkaround(realm);
+        workaround.unuseTopLevelFlowIfNeeded(topLevelFlowToImport.getAlias());
+
+        authenticationFlowRepository.deleteTopLevelFlow(realm.getRealm(), patchedAuthenticationFlow.getId());
         authenticationFlowRepository.createTopLevelFlow(realm.getRealm(), patchedAuthenticationFlow);
 
         AuthenticationFlowRepresentation createdTopLevelFlow = authenticationFlowRepository.getTopLevelFlow(realm.getRealm(), topLevelFlowToImport.getAlias());
         executionFlowsImportService.createExecutionsAndExecutionFlows(realm, topLevelFlowToImport, createdTopLevelFlow);
+
+        workaround.resetFlowIfNeeded();
+    }
+
+    /**
+     * There is no possibility to update a top-level-flow and it's not possible to recreate a top-level-flow
+     * which is currently in use. So we have to unuse our top-level-flow by use another flow temporary
+     * If no other top-level-flow can be found, this workaround will create a new one
+     * This code could be maybe replace by a better update-algorithm of top-level-flows
+     */
+    private class UsedRegistrationFlowWorkaround {
```

Comment:
Mach man das so bei Java? Klassen in Klassen, bzw. mehrere Klassen in einer Datei?